### PR TITLE
Release v3.6.0

### PR DESCRIPTION
**Pull request by joaander**
## Description

Changes necessary to release v3.6.0.

## Motivation and context

Make the latest changes available to users.

## How has this been tested?

CI checks.

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Comment by joaander** [comment]
Tests pass, but GitHub is failing to report that in the checks section of the PR: https://github.com/glotzerlab/hoomd-blue/actions/runs/3311839252

File: BUILDING.rst [hunk 1]
```diff
@@ -104,7 +104,7 @@ Install prerequisites
   - hipCUB, included for NVIDIA GPU targets, but required as an
     external dependency when building for AMD GPUs
   - roctracer-dev
-  - Linux kernel >= 3.5.0
+  - Linux kernel >= 3.6.0
```
Comment (tommy-waltmann) [inline BUILDING.rst]: Is this version supposed to be bumped? Seems to do with linux kernel version and not hoomd version.

**Review by tommy-waltmann — COMMENTED** [review]
Just one quick question.

File: BUILDING.rst (hunk 1 above)
Comment (joaander) [reply BUILDING.rst]: You are correct, I will fix. Cause: `bump2version` is a simple pattern matcher and I missed this in my review.

**Pull request merged**